dayrepo/snippets/views.py

```diff
@@ -225,117 +226,179 @@ def get(self, request, snippet_id):
-                    client=req.getlist("client")[i],
-                    goods=req.getlist("goods")[i],
-                    load_situation=req.getlist("load_situation")[i],
-                    load_mileage=req.getlist("load_mileage")[i],
-                    hollow_mileage=req.getlist("hollow_mileage")[i],
-                    is_load_situation=req.getlist("is_load_situation")[i],
-                )
-                if process.is_load_situation == "on":
-                    process.is_load_situation = True
-                else:
-                    process.is_load_situation = False
-                post_process[i].delete()
-                process.save()
-            post_snippet=snippet
-            # トップ画面へ
-            return redirect(to="snippet_list")
+                process.is_load_situation = False
+            post_process[i].delete()
+            process.save()
+        post_snippet = snippet
+        # トップ画面へ
+        return redirect(to="snippet_list")
+
 snippet_edit = SnippetEditView.as_view()
 
+
+class DbDeletePattern():
+    def snippet_del(self,target_id:int):
+        # snippetと紐づいたdutiestroubleの削除
+        dutiestrouble = DutiesTrouble.objects.get(
+            snippet_id=target_id
+        )
+        dutiestrouble.delete()
+        # snippetと紐づいたprocessの削除
+        process = get_list_or_404(
+            Process, snippet_id=target_id
+        )
+        for process_object in process:
+            process_object.delete()
+
+        snippet = Snippet.objects.get(
+            pk=target_id
+        )
+        snippet.delete()
+
+
+        checklist = Checklist.objects.get(
+            pk=snippet.checklist_id.id
+        )
+
+        # 対象のsnippetが削除されたことを反映し、
+        # checklistのみの投稿として表示、判定させる
+        checklist.is_snippet_make = False
+        checklist.save()
+
+    # 未提出postdataのchecklistを削除する
+    def checklist_del(self,target_id:int):
+        checklist = Checklist.objects.get(
+        pk=target_id
+        )
+        checklist.delete()
+
+    # 提出済みpostdataをすべて削除する
+    def all_del(self,target_id:int):
+        snippet = Snippet.objects.get(
+            pk=target_id
+        )
+        checklist = Checklist.objects.get(
+            pk=snippet.checklist_id.pk
+        )
+        dutiestrouble = DutiesTrouble.objects.get(
+            snippet_id=target_id
+        )
+        dutiestrouble.delete()
+        process_list = get_list_or_404(
+            Process, 
+            snippet_id=target_id
+        )
+        for process in process_list:
+            process.delete()
+        snippet.delete()
+        checklist.delete()
+
+def db_delete(request:HttpRequest, target_id:int, delete_type:str):
+    # 受け取ったdelete_typeから
+    # 削除対象を判別し、実行クラスの処理を選択
+    del_class = DbDeletePattern()
+    if delete_type == "snippet":
+        del_class.snippet_del(target_id)
+    elif delete_type == "checklist":
+        del_class.checklist_del(target_id)
+    elif delete_type == "all":
+        del_class.all_del(target_id)
```

Comment:
一連の if 文とこの関数の return は別の処理なので、改行したいです〜
```suggestion

        del_class.all_del(target_id)
```